### PR TITLE
[e2e] stop using deprecated framework.ExpectNotEqual in test derictory

### PR DESCRIPTION
**Pull request by bzsuni**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
stop using deprecated framework.ExpectNotEqual in test derictory

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

### CODE REVIEW

File: test/e2e/apimachinery/custom_resource_definition.go [hunk 1]
```diff
@@ -122,7 +123,7 @@ var _ = SIGDescribe("CustomResourceDefinition resources [Privileged:ClusterAdmin
 						expected = e
 					}
 				}
-				framework.ExpectNotEqual(expected, nil)
+				gomega.Expect(expected).NotTo(gomega.Equal(nil))
```
Comment (jiahuif) [inline test/e2e/apimachinery/custom_resource_definition.go:126]: There is a special ToBe(NonNil), can save a bit brain power for the reader.

**Review by pohly — CHANGES_REQUESTED** [review]
Can you do this change in all of `test/e2e` and `test/e2e_node`? Then we only need to review one PR.

**Comment by bzsuni** [comment]
> Can you do this change in all of `test/e2e` and `test/e2e_node`? Then we only need to review one PR.

Thank you for your comment. I wanted to use the directory module to divide it. Is it more organized?

**Comment by pohly** [comment]
It depends on owners, complexity and size. In this case the change is more related to SIG Testing than any individual SIG, therefore it makes more sense for me to review and approve. It's ~100 lines that need to be changed, which is manageable for a single PR.

**Comment by bzsuni** [comment]
I have made a modification, and there are some errors reported by ci, but it does not seem to be related to pr? 
If you have the time, please help me rereview this pr, thanks @pohly

**Comment by bzsuni** [comment]
confilict with other pr

**Pull request closed without merging**